scripts/tasks/size-audit.js

```diff
@@ -0,0 +1,37 @@
+const path = require('path');
+const fs = require('fs');
+
+exports.generateSizeData = function() {
+  const projectName = path.basename(process.cwd());
+  const sizeFilePath = path.join(process.cwd(), 'dist', `${projectName}.json`);
+  if (fs.existsSync(path.dirname(sizeFilePath))) {
+    const result = {
+      chunks: analyzeChunks()
+    };
+    const fileContents = JSON.stringify(result);
+    fs.writeFileSync(sizeFilePath, fileContents);
+  }
+};
+
+function analyzeChunks() {
+  const distPath = path.join(process.cwd(), 'dist');
+  const result = {};
+  for (const filePath of fs.readdirSync(distPath)) {
+    const fileName = path.basename(filePath);
+    const extension = path.extname(fileName);
+    if (extension.toLowerCase() === '.js') {
+      result[fileName] = getFileSize(path.join(distPath, fileName));
+    }
+    return result;
+  }
+}
+
+function getFileSize(filePath) {

```

Comment:
Not sure under what circumstance you wouldn't be able to get the file size. I'm thinking maybe we should be faililng the build if we can't even determine the file size here. This way, we wouldn't have a "loophole" for slipping through size issues. We probably should just `throw e` instead of return -1.

Reply:
maybe just remove the try/catch here. No need to hide and obfuscate the actual exception.